test/parallel/test-http2-compat-serverresponse-end.js

```diff
@@ -0,0 +1,42 @@
+// Flags: --expose-http2
+'use strict';
+
+const { mustCall, expectsError } = require('../common');
+const { createServer, connect } = require('http2');
+
+// Http2ServerResponse should throw when writing if the stream is closed
+
+const server = createServer();
+server.listen(0, mustCall(() => {
+  const port = server.address().port;
+  server.once('request', mustCall((request, response) => {
+    response.stream.on('finish', mustCall(() => {
+      try {
```

Comment:
Hah! Facepalm...  😳 That totally slipped my mind. This whole evening I was feeling uncomfortable about this try/catch mess. Thanks for the reminder.